sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -190,6 +190,15 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val CONSTRAINT_PROPAGATION_ENABLED = buildConf("spark.sql.constraintPropagation.enabled")
+    .internal()
```

Comment:
To determine whether a flag is internal or not, we should consider the impact of external users. If users could easily hit this, we might need to expose it as an external flag and document it in [the public document](http://spark.apache.org/docs/latest/sql-programming-guide.html).